jscomp/core/lam_compile_primitive.ml

```diff
@@ -78,6 +103,27 @@ let translate loc (cxt : Lam_compile_context.t) (prim : Lam_primitive.t)
           | _ -> E.runtime_call Js_runtime_modules.option "nullable_to_opt" args
           )
       | _ -> assert false)
+  | Pimport -> (
+      match args with
+      | [ e ] -> (
+          let output_dir = Filename.dirname output_prefix in
+
+          let module_id, module_value =
+            match module_of_expression e.expression_desc with
+            | [ module_ ] -> module_
+            | _ -> assert false
+            (* TODO: graceful error message here *)
```

Comment:
Indeed. Is this where an error occurs when one makes the mistake of trying to dynamically load a local module?
Are there other cases where this will give an error? What about if one tries to use the result of calling a functor?

Reply:
Let me try to write some obviously wrong usage.